File: webcam-dev/serial.py
import time
import serial

ser = serial.Serial(
    port='/dev/ttyUSB0',
    baudrate = 9600,
    parity=serial.PARITY_NONE,
    stopbits=serial.STOPBITS_ONE,
    bytesize=serial.EIGHTBITS,
    timeout=10
)
counter=0
while counter < 10:
    angle = 90 * counter % 2
    ser.write('steer %d'%(angle))
    ser.write('steer %d\r'%(angle))
    time.sleep(1)
    counter += 1

File: webcam-dev/serial.js
var steerRobot = function(angle){}
var speedRobot = function(angle){}
var stopRobot = function(){};

if(process.env.PI === "true"){
	console.log('starting up raspi serial.. ')
  const raspi = require('raspi');
  const Serial = require('raspi-serial').Serial;
  try {
	raspi.init(() => {
		console.log('connecting to serial port');
		var serial = new Serial({portId:"/dev/ttyUSB0", baudrate: 9600});
		serial.open(() => {
			console.log('successfully opened serial port!');
			stopRobot = function() {
				console.log('SER: speed ' + 0 + '\r');
				serial.write('speed ' + 0 + '\r');
			}
			steerRobot = function(angle) {
				console.log('SER: steer ' + clamp_value(angle, -90, 90) + '\r');
				serial.write('steer ' + clamp_value(angle, -90, 90) + '\r');
			}
			speedRobot = function(angle) {
				console.log('SER: speed ' + clamp_value(angle, -90, 90) + '\r');
				serial.write('speed ' + clamp_value(angle, -90, 90) + '\r');
			}
		});	  });
  } catch (error) {
	  console.log("ERROR connecting to serial port!");
	  console.log(error);
  }
}
